projects/button_adc/src/main.c

```diff
@@ -0,0 +1,65 @@
+#include <stdbool.h>
+#include "gpio.h"
+#include "gpio_it.h"
+#include "inc/adc.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// Interrupt callback
+static void prv_adc_callback(GpioAddress *address, void *context) {
+  GpioAddress *adc_address = context;
+
+  uint16_t adc_reading = 0;
+  adc_read_raw_pin(*adc_address, &adc_reading);
+
+  LOG_DEBUG("Read ADC value");
+}
+
+int main(void) {
+  // Address Setups
+  GpioAddress adc_address = {
+    .port = GPIO_PORT_A,  //
+    .pin = 6,             //
+  };
+
+  GpioAddress button_address = {
+    .port = GPIO_PORT_B,  //
+    .pin = 2,             //
+  };
+
+  GpioSettings gpio_settings = {
```

Comment:
You need separate settings for the ADC pin and the button pin. They should be different. One reads an analog signal. Another reads a digital one.